oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/RecoveryLockTest.java

```diff
@@ -182,6 +185,49 @@ public void selfRecoveryWithinDeadline() throws Exception {
         assertFalse(lock1.acquireRecoveryLock(2));
     }
 
+    @Test
+    // OAK-9401: Reproduce a bug that happens when the cluster is not active having a null leaseEndTime
+    public void breakRecoveryLockOfNotActiveCluster() throws Exception {
+        // Create a mocked version of the DocumentStore, to replace a method later during the test
+        DocumentStore store = spy(new MemoryDocumentStore());
+
+        info1 = ClusterNodeInfo.getInstance(store, RecoveryHandler.NOOP,
+                null, "node1", 1);
+        RecoveryLock recLock = new RecoveryLock(store, clock, 1);
+
+        // expire clusterId 1
+        clock.waitUntil(info1.getLeaseEndTime() + DEFAULT_LEASE_UPDATE_INTERVAL_MILLIS);
+        MissingLastRevSeeker seeker = new MissingLastRevSeeker(store, clock);
+
+        Semaphore recovering = new Semaphore(0);
+        Semaphore recovered = new Semaphore(0);
+        // simulate new startup and get info again
+        Future<ClusterNodeInfo> infoFuture = executor.submit(() ->
```

Comment:
Similar here. `infoFuture` is not used later on.
```suggestion
        executor.submit(() ->
```